fastlane/lib/fastlane/actions/github_api.rb

```diff
@@ -0,0 +1,266 @@
+module Fastlane
+  module Actions
+    module SharedValues
+      GITHUB_API_STATUS_CODE = :GITHUB_API_STATUS_CODE
+      GITHUB_API_RESPONSE = :GITHUB_API_RESPONSE
+      GITHUB_API_JSON = :GITHUB_API_JSON
+    end
+
+    class GithubApiAction < Action
+      def self.run(params)
+        require 'json'
+
+        http_method = (params[:http_method] || 'GET').to_s.upcase
+        url = parse_url(params[:server_url], params[:path], params[:url])
+        headers = parse_headers(params[:api_token], params[:headers])
+        request_body = parse_body(params[:body], params[:raw_body])
+        handled_errors = params[:errors] || {}
```

Comment:
yup, much better, I think I was going for it being succinct, but yeah far better to be descriptive. All usage changed to `error_handlers` and associated variable name.